sureal/tools/decorator.py

```diff
@@ -73,7 +72,7 @@ def __init__(self, func):
         self.cache = {}
 
     def __call__(self, *args):
-        if not isinstance(args, collections.Hashable):
+        if not hasattr(args, "__hash__"):
```

Comment:
This is a simple way of doing the same thing that works in all pythons.
`import collections` becomes `import collections.abc` on python3... and getting to `collections.Hashable` will error out in python 3.8